metaflow/plugins/argo/argo_workflows_deployer.py

```diff
@@ -188,6 +225,59 @@ def delete(instance: DeployedFlow, **kwargs):
     return command_obj.process.returncode == 0
 
 
+def from_deployment(identifier: str, metadata: str = None):
+    client = ArgoClient(namespace=KUBERNETES_NAMESPACE)
+    workflow_template = client.get_workflow_template(identifier)
+
+    metadata_annotations = workflow_template.get("metadata", {}).get("annotations", {})
+
+    flow_name = metadata_annotations.get("metaflow/flow_name", "")
+    parameters = json.loads(metadata_annotations.get("metaflow/parameters", {}))
+
+    # these two only exist if @project decorator is used..
+    branch_name = metadata_annotations.get("metaflow/branch_name", None)
+    project_name = metadata_annotations.get("metaflow/project_name", None)
+
+    project_kwargs = {}
+    if branch_name is not None:
+        if branch_name.startswith("prod."):
+            project_kwargs["production"] = True
+            project_kwargs["branch"] = branch_name[len("prod.") :]
+        elif branch_name.startswith("test."):
+            project_kwargs["branch"] = branch_name[len("test.") :]
+        elif branch_name == "prod":
+            project_kwargs["production"] = True
+
+    fake_flow_file_contents = generate_fake_flow_file_contents(
+        flow_name=flow_name, param_info=parameters, project_name=project_name
```

Comment:
what about the schedule decorator?

Reply:
the whole point of this fake file generation was so that the lower level API can parse user level parameters.

`@project` introduces `--branch` and `--production`, and this is why we had to add handle this..

so, I am not sure if we care about other class level parameters such as `@schedule` or perhaps `@conda_base, @pypi_base`